From 30178c6f2784783ac380dc9e7c926b50caa26cf9 Mon Sep 17 00:00:00 2001
From: Brian Delwiche <delwiche@google.com>
Date: Wed, 10 Aug 2022 05:01:58 +0000
Subject: [PATCH] Add local to pan_api.cc

Bug: 233604485
Test: Manual, connected BT and played audio
Tag: #security
Ignore-AOSP-First: Security
Change-Id: Ibb7fffccde731ecb7c12b8ddf212f02f156ffd89
(cherry picked from commit ca6cac4996a2a8c5369c646648ccbc49e29658c6)
Merged-In: Ibb7fffccde731ecb7c12b8ddf212f02f156ffd89
---
 system/stack/pan/pan_api.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/system/stack/pan/pan_api.cc b/system/stack/pan/pan_api.cc
index 0352ed4158..40b26a3aca 100644
--- a/system/stack/pan/pan_api.cc
+++ b/system/stack/pan/pan_api.cc
@@ -541,6 +541,7 @@ tPAN_RESULT PAN_WriteBuf(uint16_t handle, const RawAddress& dst,
     return PAN_FAILURE;
   }
 
+  uint16_t len = p_buf->len;
   result = BNEP_WriteBuf(pcb->handle, dst, p_buf, protocol, &src, ext);
   if (result == BNEP_IGNORE_CMD) {
     PAN_TRACE_DEBUG("PAN ignored data buf write to PANU");
@@ -552,7 +553,7 @@ tPAN_RESULT PAN_WriteBuf(uint16_t handle, const RawAddress& dst,
     return (tPAN_RESULT)result;
   }
 
-  pcb->write.octets += p_buf->len;
+  pcb->write.octets += len;
   pcb->write.packets++;
 
   PAN_TRACE_DEBUG("PAN successfully sent data buf to the PANU");
-- 
2.38.0.413.g74048e4d9e-goog

